### ISSUE TITLE
Type error when casting to a type

### ISSUE THREAD

**Issue by jpike88** [tgriesser/knex#3260]
Another type error... knex 0.17.3

```
const formFiles: TreeFile[] = await sql.knex('file')
				.join('form_log', 'form_log.formID', 'file.formID')
				.where('form_log.type', selectedCategory)
				.where('file.formID', formID)
				.select('file.*',
								'form_log.type as form_type',
								'form_log.title as form_title');
```
```
Type 'TreeFile | TreeFile[]' is not assignable to type 'TreeFile[]'.
  Type 'TreeFile' is not assignable to type 'TreeFile[]'.
```

**Comment by lorefnon** [comment]
I have not been able to reproduce this. Would you be able to provide an isolated minimal example using which this error can be reproduced ? 

When I tried testing out a similar query, rhs resolved to `any[]` which was expected, so assignment to any array in lhs should have succeeded.

**Comment by jpike88** [comment]
typescript 3.5.1

```

export interface File {
	name: string;
	type: string;
	size: number;
	fileID: string;
	url?: string;
}

export interface TreeFile extends File {
	id: number;
	url: string;
	isFolder: 1 | 0;
	childCount: number;
	dateCreated?: number;
	// alias for form_log.type after join
	formType?: string;
	formID?: string;
	// files / folder from a form are readonly
	readOnly?: boolean;
}


async () =>{
  

			const formFiles: TreeFile[] = await knex('file')
				.join('form_log', 'form_log.formID', 'file.formID')
				.join('clientGroup', 'clientGroup.id', 'form_log.clientGroupID')
				.where('file.plantHash', this.params.plantHash)
				.where('form_log.type', categoryForClientGroup)
				.where('clientGroup.label', selectedClientGroup)
				.where('file.formID', formID)
				.select('file.*',
								'form_log.type as form_type',
								'form_log.title as form_title');

}
```

**Comment by jpike88** [comment]
any ideas?

**Comment by kibertoad** [comment]
Fixed in 983f23a8b2ae4768a74f1db82161db64491858c8
